### PR TITLE
There were following issues while removing obsoleted data from database.

### PR DESCRIPTION
**Pull request by zain89**
1. Adding and removing had transaction overlap fixed that
2.Default image was also being deleted.

### CODE REVIEW

File: Sources/HiPMobile/BusinessLayer/ContentApiFetchers/BaseDataFetcher.cs [hunk 1]
```diff
@@ -87,12 +86,16 @@ public BaseDataFetcher(IRoutesBaseDataFetcher routesBaseDataFetcher, IExhibitsBa
...
                     return;
                 }
-                dataToRemoveFetcher.CleaupRemovedData();
                 if (token.IsCancellationRequested)
                 {
                     transaction.Rollback();
                 }
             }
-        }
+            await dataToRemoveFetcher.FetchDataToDelete (token);
```
Comment (cbruegg) [inline Sources/HiPMobile/BusinessLayer/ContentApiFetchers/BaseDataFetcher.cs]: Indentation looks wrong here, can you go through the files and hit CTRL-K and CTRL-D consecutively to auto format?

File: Sources/HiPMobile/BusinessLayer/ContentApiFetchers/DataToRemoveFetcher.cs [hunk 2]
```diff
@@ -168,7 +168,7 @@ private void RemoveRouteTags(Route route, List<RouteTag> deletedTags, List<Image
                 {
                     deletedTags.Add(tag);
                 }
-                if (tag.Image != null && !allMedias.Contains(tag.Image.IdForRestApi))
+                if (tag.Image != null && !allMedias.Contains(tag.Image.IdForRestApi) && tag.Image.IdForRestApi != -2)
```
Comment (cbruegg) [inline Sources/HiPMobile/BusinessLayer/ContentApiFetchers/DataToRemoveFetcher.cs]: What's that magic `-2` value? Perhaps put it into a named constant?

File: Sources/HiPMobile/ServiceAccessLayer/ServerEndpoints.cs [hunk 3]
```diff
@@ -26,7 +26,7 @@ public class ServerEndpoints
         /// <summary>
         /// Urlpath for the docker container running the HiP-DataStore instance
         /// </summary>
-        public const string DatastoreApiPath = "https://docker-hip.cs.uni-paderborn.de/public/datastore/api";
+        public const string DatastoreApiPath = "https://docker-hip.cs.uni-paderborn.de/develop/datastore/api";
```
Comment (cbruegg) [inline Sources/HiPMobile/ServiceAccessLayer/ServerEndpoints.cs]: Why is this change necessary? Looks important enough it would deserve its own PR, since until now we've developed against the public API that doesn't change as frequently as the develop API.

Comment (zain89) [reply Sources/HiPMobile/ServiceAccessLayer/ServerEndpoints.cs]: My bad this should be public, For testing, I was using develop. (Although this can only be tested with develop datastore)

File: Sources/HiPMobile/BusinessLayer/ContentApiFetchers/DataToRemoveFetcher.cs (hunk 2 above)
Comment (zain89) [reply Sources/HiPMobile/BusinessLayer/ContentApiFetchers/DataToRemoveFetcher.cs]: This is default image having id -2, which should not be deleted as it's not part of the data store. Will put this into constant
